controllers/eventsource/resource.go

```diff
@@ -397,6 +398,16 @@ func buildService(args *AdaptorArgs) (*corev1.Service, error) {
 			Selector:  args.Labels,
 		},
 	}
+
+	if args.EventSource.Spec.Service.Metadata != nil {
+		if args.EventSource.Spec.Service.Metadata.Labels != nil {
+			svc.ObjectMeta.SetLabels(args.EventSource.Spec.Service.Metadata.Labels)
```

Comment:
Instead of `SetLabels` and `SetAnnotations`, can we do for loops and copy the labels and annotations over? `Set` means original labels and annotations will be replaced.